pkg/apis/autoscaling/v1/conversion.go

```diff
@@ -339,47 +356,40 @@ func Convert_v1_HorizontalPodAutoscaler_To_autoscaling_HorizontalPodAutoscaler(i
 
 	if otherMetricsEnc, hasOtherMetrics := out.Annotations[autoscaling.MetricSpecsAnnotation]; hasOtherMetrics {
 		var otherMetrics []autoscalingv1.MetricSpec
-		if err := json.Unmarshal([]byte(otherMetricsEnc), &otherMetrics); err != nil {
-			return err
-		}
-
-		// the normal Spec conversion could have populated out.Spec.Metrics with a single element, so deal with that
-		outMetrics := make([]autoscaling.MetricSpec, len(otherMetrics)+len(out.Spec.Metrics))
-		for i, metric := range otherMetrics {
-			if err := Convert_v1_MetricSpec_To_autoscaling_MetricSpec(&metric, &outMetrics[i], s); err != nil {
-				return err
+		if err := json.Unmarshal([]byte(otherMetricsEnc), &otherMetrics); err == nil {
+			// the normal Spec conversion could have populated out.Spec.Metrics with a single element, so deal with that
+			outMetrics := make([]autoscaling.MetricSpec, len(otherMetrics)+len(out.Spec.Metrics))
+			for i, metric := range otherMetrics {
+				if err := Convert_v1_MetricSpec_To_autoscaling_MetricSpec(&metric, &outMetrics[i], s); err != nil {
+					return err
+				}
 			}
+			if out.Spec.Metrics != nil {
+				outMetrics[len(otherMetrics)] = out.Spec.Metrics[0]
```

Comment:
we are guaranteed that out.Spec.Metrics is non empty?  I guess your test covers that?  It's weird to assume out.Spec.Metrics has length 1 without a special guard to me.  Non obvious in code.

Reply:
this is unchanged, see https://github.com/kubernetes/kubernetes/pull/89963/files?w=1 for the changes without whitespace

Reply:
Convert_v1_HorizontalPodAutoscalerSpec_To_autoscaling_HorizontalPodAutoscalerSpec appears to populate out.Metrics with a single item list when TargetCPUUtilizationPercentage is non-nil. Will check what happens when it is nil.

Reply:
yeah, `out.Spec.Metrics` is nil unless populated by `Convert_v1_HorizontalPodAutoscalerSpec_To_autoscaling_HorizontalPodAutoscalerSpec`

Reply:
ok.